subsys/net/lib/app/server.c

```diff
@@ -311,11 +357,11 @@ static void tls_server_handler(struct net_app_ctx *ctx,
 			ctx->cb.close(ctx, -ESHUTDOWN, ctx->user_data);
 		}
 
-		if (ctx->server.net_ctx) {
-			NET_DBG("Server context %p removed",
-				ctx->server.net_ctx);
-			net_context_put(ctx->server.net_ctx);
-			ctx->server.net_ctx = NULL;
+		net_ctx = find_net_ctx(ctx, &ret);
```

Comment:
minor nit: using the ret var here is rather confusing from a readability perspective.